internal/engine/session_integration_test.go

```diff
@@ -584,50 +617,31 @@ func TestNewOrchestraClientMaybeLookupBackendsFailure(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skip test in short mode")
 	}
+	errMocked := errors.New("mocked error")
 	sess := newSessionForTestingNoLookups(t)
-	ctx, cancel := context.WithCancel(context.Background())
-	cancel() // fail immediately
-	client, err := sess.NewOrchestraClient(ctx)
-	if !errors.Is(err, ErrAllProbeServicesFailed) {
-		t.Fatal("not the error we expected")
+	sess.testMaybeLookupBackendsContext = func(ctx context.Context) error {
+		return errMocked
+	}
+	client, err := sess.NewOrchestraClient(context.Background())
+	if !errors.Is(err, errMocked) {
+		t.Fatal("not the error we expected", err)
 	}
 	if client != nil {
 		t.Fatal("expected nil client here")
 	}
 }
 
-type httpTransportThatSleeps struct {
-	txp netx.HTTPRoundTripper
-	st  time.Duration
-}
-
-func (txp httpTransportThatSleeps) RoundTrip(req *http.Request) (*http.Response, error) {
-	resp, err := txp.txp.RoundTrip(req)
-	time.Sleep(txp.st)
-	return resp, err
-}
-
-func (txp httpTransportThatSleeps) CloseIdleConnections() {
-	txp.txp.CloseIdleConnections()
-}
-
 func TestNewOrchestraClientMaybeLookupLocationFailure(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skip test in short mode")
 	}
+	errMocked := errors.New("mocked error")
 	sess := newSessionForTestingNoLookups(t)
-	sess.httpDefaultTransport = httpTransportThatSleeps{
-		txp: sess.httpDefaultTransport,
-		st:  5 * time.Second,
-	}
-	// The transport sleeps for five seconds, so the context should be expired by
-	// the time in which we attempt at looking up the location. Because the
-	// implementation performs the round-trip and _then_ sleeps, it means we'll
-	// see the context expired error when performing the location lookup.
-	ctx, cancel := context.WithTimeout(context.Background(), 4*time.Second)
-	defer cancel()
-	client, err := sess.NewOrchestraClient(ctx)
-	if !errors.Is(err, geolocate.ErrAllIPLookuppersFailed) {
+	sess.testMaybeLookupLocationContext = func(ctx context.Context) error {
+		return errMocked
+	}
+	client, err := sess.NewOrchestraClient(context.Background())
+	if !errors.Is(err, errMocked) {
```

Comment:
This test needed adjusting because now MaybeLookupLocationContext immediately fails w/ a cancelled context